src/pages/video-watch/VideoWatch.jsx

```diff
@@ -0,0 +1,252 @@
+    const {postData, deleteData} = useAxios();
+    console.log({dataState});
+    const {loggedInUser} = useAuth();
+    const allVideos = dataState.allVideos;
+    const userPlaylists = dataState.userPlaylists
+    const video = getVideoDetails(allVideos,videoId)    
+
+
+    const addToLikedVideos = async () => { 
+        setIsLikedLoading(true);
+        await addToPlaylistVideos(loggedInUser,video,getUserPlaylist(userPlaylists,'Liked','default'),dataDispatch,postData); 
+        const likes = video.likes + 1;
+        await postData(`/users/${video.user._id}/videos/${video._id}`,{likes })
+        dataDispatch({type:UPDATE_USER_VIDEO, payload:{            
+            video,
+            key : 'likes',
+            value : likes
+        }})
+        setIsLikedLoading(false);    
+    }
+    
+    
+    const removeFromLikedVideos = async () => { 
+        setIsLikedLoading(true);        
+        await removeFromPlaylistVideos(loggedInUser,getUserPlaylistVideo(video,getUserPlaylist(userPlaylists,'Liked','default')),getUserPlaylist(userPlaylists,'Liked','default'),dataDispatch,deleteData)
+        const likes = video.likes - 1;
+        await postData(`/users/${video.user._id}/videos/${video._id}`,{likes })
+        dataDispatch({type:UPDATE_USER_VIDEO, payload:{            
+            video,
+            key : 'likes',
+            value : likes
+        }})
+        setIsLikedLoading(false);    
+    }
+
+
+    const addToWatchLaterVideos = async() => {
+        setIsWatchLaterLoading(true);
+        await addToPlaylistVideos(loggedInUser,video,getUserPlaylist(userPlaylists,'Watch Later','default'),dataDispatch,postData); 
+        setIsWatchLaterLoading(false);
+    }
+
+    const removeFromWatchLaterVideos = async() => {
+        setIsWatchLaterLoading(true);
+        await removeFromPlaylistVideos(loggedInUser,getUserPlaylistVideo(video,getUserPlaylist(userPlaylists,'Watch Later','default')),getUserPlaylist(userPlaylists,'History','default'),dataDispatch,deleteData)
+        setIsWatchLaterLoading(false);
+    }
+
+    const togglePlaylistModal = () => {
+        playlistModalRef.current.classList.toggle('active')
+    }  
+
+
+    useEffect(() => {        
+        (async function(){
+            await postData(`/users/${video.user._id}/videos/${video._id}`,{views : video.views+1})
+            dataDispatch({type:UPDATE_USER_VIDEO, payload:{            
+                video,
+                key : 'views',
+                value : video.views + 1
+            }})
+        })()
+        if(loggedInUser){
+            (async function(){
+                await addToPlaylistVideos(loggedInUser,video,getUserPlaylist(userPlaylists,'History','default'),dataDispatch,postData);
+            })()
+        }      
+        // eslint-disable-next-line  
+    },[])
+
+    
+    return (
+        <>                   
+            <div className="video-container">            
+                <iframe width="100%" height="300px" 
```

Comment:
Instead of using `<iframe>` try [ReactPlayer](https://www.npmjs.com/package/react-player), Just a suggestion.